### DISCUSSION TITLE
Building a REST API on top of a database

### DISCUSSION THREAD

**Discussion by prrao87** [thedataquarry/rustinpieces#21] [Ideas]
I think a really useful piece to add to the APIs section would be an example FastAPI web server on top of Postgres (the same data from the `postgres_etl` piece) for Python, and an Axum web server on top of Postgres for Rust.

I think among all the backend web frameworks out there for Rust, Axum has gotten the most traction recently, and it would be interesting to build in it for someone coming with FastAPI knowledge in Python.

**Comment by prrao87** [comment]
I'm open to other ideas on APIs that can serve other needs (i.e., not being built on top of a DB) as well. But I feel like a lot of data engineers and software engineers coming from Python would be interested in interfacing with DBs, which is why that would be a useful first exercise.

**Comment by sanders41** [comment]
I have had this exact topic on my to-do list to look into for months, and Axum is the framework I had in mind also. With web being one of the biggest uses of Python I imagine there would be a lot of people interested in this topic.
